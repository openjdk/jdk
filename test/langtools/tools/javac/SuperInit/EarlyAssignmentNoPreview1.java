/*
 * @test /nodynamiccopyright/
 * @bug 8334258
 * @summary Disallow early assignment if FLEXIBLE_CONSTRUCTORS preview feature is not enabled
 * @compile/fail/ref=EarlyAssignmentNoPreview1.out -XDrawDiagnostics EarlyAssignmentNoPreview1.java
 */
public class EarlyAssignmentNoPreview1 {

    Runnable r;

    public EarlyAssignmentNoPreview1() {
        this(r = () -> System.out.println("hello"));
    }

    public EarlyAssignmentNoPreview1(Runnable r) {
    }

    public static void main(String[] args) {
        new EarlyAssignmentNoPreview1();
    }
}
