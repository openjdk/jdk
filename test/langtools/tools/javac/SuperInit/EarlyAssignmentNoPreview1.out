EarlyAssignmentNoPreview1.java:12:14: compiler.err.preview.feature.disabled: (compiler.misc.feature.flexible.constructors)
1 error
